text/src/swam/witx/unresolved/Types.scala

```diff
@@ -0,0 +1,36 @@
+package swam
+package witx
+package unresolved
+
+/**
+@author Javier Cabrera-Arteaga on 2020-03-18
+  */
+sealed trait BaseWitxType
+
+case class BasicType(name: String) extends BaseWitxType
```

Comment:
These types are a known finite list of possibility? If yes, make it a sealed trait with all possible values as objects (enum style). This will catch a lot of problem later in pattern matching.

Reply:
I think the WITX specification is not definitive, But I will seal these objects